packages/api-bindings/src/graphql/search.graphql

```diff
@@ -0,0 +1,63 @@
+query SearchPublications(
+  $limit: LimitScalar
+  $cursor: Cursor
+  $query: Search!
+  $sources: [Sources!]
+  $customFilters: [CustomFiltersTypes!]
+  $observerId: ProfileId
+) {
+  result: search(
+    request: {
+      query: $query
+      type: PUBLICATION
+      limit: $limit
+      cursor: $cursor
+      sources: $sources
+      customFilters: $customFilters
+    }
```

Comment:
Let's reduce the amount of translations by exposing just the `request` object to our hook code.

Reply:
I think this is not possible here without some hacky solution, if we just expose `request` then we need to make a generic `search` query for both `SearchPublications` and  `SearchProfiles`, because we can't set up `type: PUBLICATION`  or `type:PROFILE` directly in the query:

```graphql
query SearchPublications(
  $request: SearchRequestQuery
  $observerId: ProfileId
) {
  result: search( request: $request) {
    ... on PublicationSearchResult {
      __typename
      items {
        ... on Post {
          ...Post
        }

        ... on Comment {
          ...Comment
        }
      }
      pageInfo {
        ...CommonPaginatedResultInfo
      }
    }
  }
  
      ... on ProfileSearchResult {
      __typename
      items {
        ...ProfileFields
      }
      pageInfo {
        ...CommonPaginatedResultInfo
      }
    }
}
```

and as we know this throws an error because the type from `id` is different in both `Profile` and `Publication`
I tried to find some way to do a spread or setup a default parameter in a graphql query but i found nothing, do you have any suggestion?

Reply:
Well, we can probably set the `type` on the react hook, but then the query defined in `search.graphql` as `SearchPublications` will have no sense outside of the react hook context